src/SMTLIB/Backends/Process.hs

```diff
@@ -93,6 +97,12 @@ new config logger = do
             IO.hClose h `X.catch` \ex ->
               logger $ BS.pack $ show (ex :: X.IOException)
           )
+    logger = logText . decodeUtf8Lenient
```

Comment:
I imagine `logText` is going to immediately reencode the Text to print it. Is the Text interface serving any purpose?

Also, are you sure the encoding will be UTF-8 in every system?

Reply:
> I imagine logText is going to immediately reencode the Text to print it. Is the Text interface serving any purpose?

The point of the `Text` interface is to have a common denominator to the logging functions. It's not as efficient as exposing the raw types but should make it simpler to the user. Since logging doesn't need to be particularly efficient I figured it'd be fine to make this compromise here.

> Also, are you sure the encoding will be UTF-8 in every system?

The [SMT-LIB standard](https://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.6-r2021-05-12.pdf#part.2) says:
> SMT-LIB source files consist of Unicode characters in any 8-bit encoding, such as UTF-8,
that extends the original 7-bit US-ASCII set. While not technically Unicode, the ISO 8859-1
character set is also allowed since it coincides with the first 256 characters of UTF-8.

So I figured it was reasonable to use UTF-8 as the de-facto standard.

---

I guess I could replace `Text`s with bytestrings as the common input type to the logger functions. I'm not to keen on using `Builder`s because giving a basic printing function is not super simple: there is no standard `putBuilder :: Builder -> IO ()` in the API.

Reply:
Looks the safest to me too, and it was unclear how much the user wins from getting the messages converted to Text early.